src/requests/typings.ts

```diff
@@ -0,0 +1,38 @@
+import * as runtypes from 'runtypes'
+
+const MethodValidator = runtypes.Union(
+  runtypes.Literal('GET'),
+  runtypes.Literal('PATCH'),
+  runtypes.Literal('HEAD'),
+  runtypes.Literal('POST'),
+  runtypes.Literal('DELETE'),
+  runtypes.Literal('OPTIONS'),
+  runtypes.Literal('PUT')
+)
+
+const PathValidator = runtypes.String.withConstraint((s) => s.startsWith('/'), {
+  name: 'CanonicalURI',
+})
+
+export const CanonicalRequestValidator = runtypes
+  .Record({
+    method: MethodValidator,
+    path: PathValidator,
+  })
+  .And(
+    runtypes.Partial({
```

Comment:
How about requiring both (and expecting `{}` and empty string `''` respectively)

Reply:
I agree for `headers`, but I am not really sure about the `body`. For a GET request wouldn't feel awkward having to specify a `body` (even if it's empty string)?